controllers/provider-alicloud/pkg/webhook/controlplanebackup/ensurer.go

```diff
@@ -64,6 +75,15 @@ func (e *ensurer) ensureContainers(ps *corev1.PodSpec, name string, cluster *ext
 	return nil
 }
 
+func (e *ensurer) ensureChecksumAnnotations(ctx context.Context, template *corev1.PodTemplateSpec, namespace, name string) error {
+	if name == common.EtcdMainStatefulSetName {
+		if err := controlplane.EnsureSecretChecksumAnnotation(ctx, template, e.client, namespace, alicloud.BackupSecretName); err != nil {
```

Comment:
Who is actually adding this secret volume to the `StatefulSet` spec? Is this missing at all? I guess Gardener cannot add the secret by default. In the future, the backup extension controller will create the secret anyways (and not Gardener anymore).

Similarly, I think Gardener should not add + mount the client certificate secret anymore, right? It's only needed by the backup-restore sidecar which we are injecting. So, this is still missing (apart from the PR), is this assumption correct?

Reply:
In this particular case the `etcd-backup` secret is not mounted as a volume but is still used as a source of env variables, injected by the webhook, which warrants adding a checksum annotations (but only for `etcd-main`, not `etcd-events`)
Regarding the client certificate secret `etcd-client-tls`, I see it currently mounted to the `etcd` container as well. Do you mean that it's not needed there, or do you mean a different secret (which one)? Looking at the chart in Gardener it seems to me that the only secret that  is not added by Gardener is the `etcd-backup` secret.

Reply:
OK, thank you, makes sense!

> Do you mean that it's not needed there, or do you mean a different secret (which one)?
It's not needed I think (@swapnilgm can you confirm that, please?) - at least, I don't see it anywhere used by etcd itself.

Reply:
> I think Gardener should not add + mount the client certificate secret anymore, right?

Its needed in etcd container for it's liveness probe. So, it should come from Gardener.

Regarding  etcd-backup secret, based on the cloud provider its either mounted or referenced in environment variable. It is required only by backup sidecar. Though currently Gardener creates it. In future  backup extension controller will take responsibility to deploy secret for etcd.

Reply:
Ah okay, right, thanks for pointing this out!